File: app/src/main/java/vgame/ir/exoplayer/DescriptionAdapter.java
package vgame.ir.exoplayer;

import android.app.PendingIntent;
import android.content.Context;
import android.graphics.Bitmap;
import android.net.Uri;

import androidx.annotation.Nullable;

import com.bumptech.glide.Glide;
import com.google.android.exoplayer2.Player;
import com.google.android.exoplayer2.ui.PlayerNotificationManager;

import java.util.concurrent.ExecutionException;

public class DescriptionAdapter implements
        PlayerNotificationManager.MediaDescriptionAdapter {

    private String title, content, url;
    private Context context;
    private PendingIntent pendingIntent;

    public DescriptionAdapter(Context context, String title, String content, String url ,PendingIntent pendingIntent) {
        this.context = context;
        this.content = content;
        this.title = title;
        this.url = url;
        this.pendingIntent = pendingIntent;
    }

    @Override
    public String getCurrentContentTitle(Player player) {
        return title;
    }

    @Nullable
    @Override
    public String getCurrentContentText(Player player) {
        return content;
    }

    @Nullable
    @Override
    public Bitmap getCurrentLargeIcon(Player player,
                                      PlayerNotificationManager.BitmapCallback callback) {
        Uri uri = Uri.parse(url);
        Bitmap notification_artwork = null;

        try {
            notification_artwork = Glide.with(context)
                    .asBitmap()
                    .load(uri)
                    .submit(100, 100).get();
        } catch (Throwable t) {
            if (t instanceof ExecutionException) {
                t = t.getCause();
            }
        }
        return notification_artwork;
    }

    @Nullable
    @Override
    public PendingIntent createCurrentContentIntent(Player player) {
        return pendingIntent;
    }
}

File: app/src/main/java/vgame/ir/exoplayer/DescriptionAdapter.kt
package vgame.ir.exoplayer

import android.app.PendingIntent
import android.content.Context
import android.graphics.Bitmap
import android.net.Uri

import com.bumptech.glide.Glide
import com.google.android.exoplayer2.Player
import com.google.android.exoplayer2.ui.PlayerNotificationManager

import java.util.concurrent.ExecutionException

class DescriptionAdapter(private val context: Context, private val title: String, private val content: String, private val url: String, private val pendingIntent: PendingIntent) : PlayerNotificationManager.MediaDescriptionAdapter {

    override fun getCurrentContentTitle(player: Player): String {
        return title
    }

    override fun getCurrentContentText(player: Player): String? {
        return content
    }

    override fun getCurrentLargeIcon(player: Player,
                                     callback: PlayerNotificationManager.BitmapCallback): Bitmap? {
        val uri = Uri.parse(url)
        var notification_artwork: Bitmap? = null

        try {
            notification_artwork = Glide.with(context)
                    .asBitmap()
                    .load(uri)
                    .submit(100, 100).get()
        } catch (t: Throwable) {
            if (t is ExecutionException) {
                t = t.cause
            }
        }

        return notification_artwork
    }

    override fun createCurrentContentIntent(player: Player): PendingIntent? {
        return pendingIntent
    }
}
